### DISCUSSION TITLE
DropZone / FileTrigger component - visualising progress of file upload

### DISCUSSION THREAD

**Discussion by isaakdc** [adobe/react-spectrum#6879] [Q&A]
Hi all,

As the title suggests I'm looking for a way of visualising the upload progress using the DropZone / FileTrigger component. How would you go about getting the upload progress and possibly other information on the file using these components?

**Comment by reidbarber** [comment]
Upload progress is outside the scope of those components, but you can use the XMLHttpRequestUpload [progress event](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequestUpload/progress_event#using_the_progress_event) to update the state of a [ProgressBar](https://react-spectrum.adobe.com/react-aria/ProgressBar.html).

For the file information, you can take a look at the [FileDropItem](https://react-spectrum.adobe.com/react-aria/useDrop.html#files) type.
